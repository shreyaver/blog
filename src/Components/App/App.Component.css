/* .App {
    display: flex;
    flex-direction: column;
    width: 100%;
    justify-content: space-between;
}  */
.Button-div {
    text-align: center;
}