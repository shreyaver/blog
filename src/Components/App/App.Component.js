import React, { Component } from 'react';
import './App.Component.css';
import AllPosts from '../AllPosts/AllPosts.Component';
import Header from '../Header/Header.Component';
import Button from '../Button/Button.Component';
import Footer from '../Footer/Footer.Component';
import getRequest from '../../Helpers/Request/request.js';
import LoadingPage from '../LoadingPage/LoadingPage.Component.js';

class App extends Component {
  state = {
    cardDataArr: []
  }
  componentDidMount() {
    getRequest('https://api.myjson.com/bins/hc5ye').then((cardDataArr) => {
      this.setState({ cardDataArr: cardDataArr.data});
    });
  }
  render() {
    if (this.state.cardDataArr.length === 0) {
      return (<LoadingPage />);
    } else {
    return (
      <div className = "App">
        <Header />
        <AllPosts cardDataArr = {this.state.cardDataArr}/>
        <div className="Button-div">
          <Button type="button" caption = "CREATE POST" onClick={() => window.location.href="/post"}/> 
        </div>
        <Footer />
      </div>
    );
  }
}
}

export default App;
