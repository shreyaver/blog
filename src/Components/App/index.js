import React from 'react';
import './App.css';
import AllPosts from '../AllPosts';
import Header from '../Header';
import Button from '../Button';
import Footer from '../Footer';
import { Link } from 'react-router-dom';

const App = (props) => {
  return (
    <div className = "App">
      <Header />
      <AllPosts cardDataArr = {props.cardDataArr}/>
      <div className="Button-div">
        <Button type="button" caption = "CREATE POST" onClick={() => window.location.href="/post"}/> 
      </div>
      <Footer />
    </div>
  );
}

export default App;
