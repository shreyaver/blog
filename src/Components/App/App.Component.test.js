import React from 'react';
import renderer from 'react-test-renderer';
import App from './App.Component';
import { shallow } from 'enzyme';
import mockPosts from '../../mockData';
import axios from 'axios';

describe('App', () => {
  let getMock;
  beforeAll(() => {
    getMock = jest.spyOn(axios, 'get');
    getMock.mockImplementation(() => {
      return Promise.resolve({ 'data': mockPosts })
    });
  });
  afterAll(() => {
    getMock.mockRestore();
  });
  it('renders without crashing for loading page', () => {
    const app = renderer.create(<App />).toJSON();
    expect(app).toMatchSnapshot();
  });
  it('renders without crashing after data is fetched', async () => {
    const app = renderer.create(<App />);
    await app.toTree().instance.componentDidMount();
    expect(app.toJSON()).toMatchSnapshot();
  });
  it('sets initial state', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.instance().state.cardDataArr).toEqual([]);
  });
  it('sets state to fetched data', async (done) => {
    const wrapper = shallow(<App />);
    await wrapper.instance().componentDidMount();
    expect(wrapper.instance().state.cardDataArr).toEqual(mockPosts);
    done();
  });
});

