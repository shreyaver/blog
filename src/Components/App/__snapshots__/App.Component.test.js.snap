// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing after data is fetched 1`] = `
<div
  className="App"
>
  <header>
    <div
      className="Content"
    >
      <div>
        <h1>
          The Artifact
        </h1>
        <h2>
          Culture & Art blog
        </h2>
      </div>
      <nav>
        <div
          className="navBar"
        >
          <a
            href="/"
          >
            Blog
          </a>
          <a
            href="#"
          >
            About
          </a>
          <a
            href="#"
          >
            Contact
          </a>
        </div>
      </nav>
    </div>
  </header>
  <div
    className="All-posts"
  >
    <div
      className="Blog-post"
    >
      <div
        className="Post-content"
      >
        <div
          className="Image-panel"
          style={
            Object {
              "backgroundImage": "url(abstract.png)",
            }
          }
        />
        <article
          className="Post-text"
        >
          <div>
            <span
              className="Date-posted"
            >
              2nd Januray, 2018
            </span>
            <span
              className="Reading-time"
            >
              2 mins
            </span>
          </div>
          <label>
            The future of abstract art and the culture ...
          </label>
          <p>
            Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
          </p>
        </article>
      </div>
      <div
        className="Post-reaction"
      >
        <hr />
        <span>
          <button
            className="Clap-icon"
            onClick={[Function]}
          />
          10
          <button
            className="Heart-icon"
            onClick={[Function]}
          />
        </span>
      </div>
    </div>
    <div
      className="Blog-post"
    >
      <div
        className="Post-content"
      >
        <div
          className="Image-panel"
          style={
            Object {
              "backgroundImage": "url(outdoor.png)",
            }
          }
        />
        <article
          className="Post-text"
        >
          <div>
            <span
              className="Date-posted"
            >
              31st Januray, 2018
            </span>
            <span
              className="Reading-time"
            >
              1 min
            </span>
          </div>
          <label>
            How painting outdoors can transform your 
          </label>
          <p>
            Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
          </p>
        </article>
      </div>
      <div
        className="Post-reaction"
      >
        <hr />
        <span>
          <button
            className="Clap-icon"
            onClick={[Function]}
          />
          1
          <button
            className="Liked-heart-icon"
            onClick={[Function]}
          />
        </span>
      </div>
    </div>
    <div
      className="Blog-post"
    >
      <div
        className="Post-content"
      >
        <div
          className="Image-panel"
          style={
            Object {
              "backgroundImage": "url(young-painters.png)",
            }
          }
        />
        <article
          className="Post-text"
        >
          <div>
            <span
              className="Date-posted"
            >
              20th March, 2018
            </span>
            <span
              className="Reading-time"
            >
              2 mins
            </span>
          </div>
          <label>
            10 young painters you need to know
          </label>
          <p>
            Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
          </p>
        </article>
      </div>
      <div
        className="Post-reaction"
      >
        <hr />
        <span>
          <button
            className="Clap-icon"
            onClick={[Function]}
          />
          15
          <button
            className="Heart-icon"
            onClick={[Function]}
          />
        </span>
      </div>
    </div>
    <div
      className="Blog-post"
    >
      <div
        className="Post-content"
      >
        <div
          className="Image-panel"
          style={
            Object {
              "backgroundImage": "url(perfect-art.png)",
            }
          }
        />
        <article
          className="Post-text"
        >
          <div>
            <span
              className="Date-posted"
            >
              22nd June, 2018
            </span>
            <span
              className="Reading-time"
            >
              5 mins
            </span>
          </div>
          <label>
            Tips for creating the perfect art
          </label>
          <p>
            Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
          </p>
        </article>
      </div>
      <div
        className="Post-reaction"
      >
        <hr />
        <span>
          <button
            className="Clap-icon"
            onClick={[Function]}
          />
          11
          <button
            className="Heart-icon"
            onClick={[Function]}
          />
        </span>
      </div>
    </div>
    <div
      className="Blog-post"
    >
      <div
        className="Post-content"
      >
        <div
          className="Image-panel"
          style={
            Object {
              "backgroundImage": "url(inspired.png)",
            }
          }
        />
        <article
          className="Post-text"
        >
          <div>
            <span
              className="Date-posted"
            >
              4th Feburary, 2019
            </span>
            <span
              className="Reading-time"
            >
              9 mins
            </span>
          </div>
          <label>
            Stay inspired: top artists blog ...
          </label>
          <p>
            Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
          </p>
        </article>
      </div>
      <div
        className="Post-reaction"
      >
        <hr />
        <span>
          <button
            className="Clap-icon"
            onClick={[Function]}
          />
          45
          <button
            className="Liked-heart-icon"
            onClick={[Function]}
          />
        </span>
      </div>
    </div>
    <div
      className="Blog-post"
    >
      <div
        className="Post-content"
      >
        <div
          className="Image-panel"
          style={
            Object {
              "backgroundImage": "url(reason.png)",
            }
          }
        />
        <article
          className="Post-text"
        >
          <div>
            <span
              className="Date-posted"
            >
              10th Januray, 2019
            </span>
            <span
              className="Reading-time"
            >
              2 mins
            </span>
          </div>
          <label>
            5 reasons why everyone should try an...
          </label>
          <p>
            Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
          </p>
        </article>
      </div>
      <div
        className="Post-reaction"
      >
        <hr />
        <span>
          <button
            className="Clap-icon"
            onClick={[Function]}
          />
          11
          <button
            className="Heart-icon"
            onClick={[Function]}
          />
        </span>
      </div>
    </div>
  </div>
  <div
    className="Button-div"
  >
    <button
      className="Button"
      onClick={[Function]}
      type="button"
    >
      CREATE POST
    </button>
  </div>
  <footer>
    <div
      className="footerContent"
    >
      <small>
        ©artifact.com 2019
      </small>
    </div>
  </footer>
</div>
`;

exports[`App renders without crashing for loading page 1`] = `
<div>
  <header>
    <div
      className="Content"
    >
      <div>
        <h1>
          The Artifact
        </h1>
        <h2>
          Culture & Art blog
        </h2>
      </div>
      <nav>
        <div
          className="navBar"
        >
          <a
            href="/"
          >
            Blog
          </a>
          <a
            href="#"
          >
            About
          </a>
          <a
            href="#"
          >
            Contact
          </a>
        </div>
      </nav>
    </div>
  </header>
  <div
    className="Loading-page"
  >
    Loading...
  </div>
  <footer>
    <div
      className="footerContent"
    >
      <small>
        ©artifact.com 2019
      </small>
    </div>
  </footer>
</div>
`;
